pybot/endpoints/slack/utils/event_utils.py

```diff
@@ -103,3 +106,69 @@ def build_messages(user_id) -> Tuple[Message, Message, Message, Message]:
         data = await response.json()
         headers = {"Authorization": f"Bearer {data['token']}"}
     return headers
+
+
+async def get_profile_suggestions(
+    slack_id: int,
+    auth_header: Dict[str, str],
+    slack_api: SlackAPI,
+    session: ClientSession,
+) -> List[Tuple[str, str]]:
+    """
+    Generates a list of slack channels to suggest
+
+    :param slack_id: The slack_id to return profile information from
+    :return: A list of slack channel names to suggest for the user
+    """
+    results = []
+
+    # Get profile data that we want to build suggestions from.
+    user_info = await slack_api.query(methods.USERS_INFO, {"user": slack_id})
+    email = user_info["user"]["profile"]["email"]
+
+    async with session.get(
+        f"{BACKEND_URL}/auth/profile/admin/",
+        headers=auth_header,
+        params={"email": email},
+    ) as response:
+        data = await response.json()
+        logger.info(f"Retrieved profile data from email: {email}")
+
+    # Get the names of all public, non-archived channels
+    channels = await slack_api.query(
+        methods.CONVERSATION_LIST, {"exclude_archived": True}
+    )
+
+    # Get a list of all the profile data we want to compare
+    possible_suggestions = parse_suggestions_from_profile(data)
+
+    # For each value in the profile data, check that we have a channel name that matches.
+    for info in possible_suggestions:
```

Comment:
Not to worried about a double for loop but you could avoid the lookup if you converted the list of channels into a set, or a key value pair. 

But I think the naming should be clearer:

```python
for suggestion in possible_suggestions:
    for channel in channels:
        if suggestion in channel[1]:
            results.append((channel[0], channel[1]))
            break
```